core/templates/pages/exploration-editor-page/translation-tab/audio-translation-bar/audio-translation-bar.directive.ts

```diff
@@ -409,12 +402,13 @@ angular.module('oppia').directive('audioTranslationBar', [
             var audioTranslation = getAvailableAudio(
               $scope.contentId, $scope.languageCode);
             if (audioTranslation) {
-              $q.when(AudioPlayerService.load(audioTranslation.filename))
+              (AudioPlayerService.loadAsync(audioTranslation.filename))
                 .then(function() {
                   $scope.audioLoadingIndicatorIsShown = false;
                   $scope.audioIsLoading = false;
                   $scope.audioTimerIsShown = true;
                   AudioPlayerService.play();
+                  $scope.$applyAsync();
```

Comment:
I'm not sure why we are adding `scope.$applyAsync` everywhere in the codebase now :( For how long is this structure going to exist?

Reply:
Async stuff right? This is not a $q thing... so angularjs doesn't track this promise. Need to force a digest cycle. This gets removed during the migration.